src/Cli/dotnet/Commands/Pack/PackCommand.cs

```diff
@@ -40,16 +42,68 @@ public static PackCommand FromParseResult(ParseResult parseResult, string? msbui
             CommonOptions.PropertiesOption,
             CommonOptions.RestorePropertiesOption,
             PackCommandParser.TargetOption);
+
         return new PackCommand(
             parsedMSBuildArgs.CloneWithAdditionalProperties(projectLocator.GetCustomDefaultConfigurationValueIfSpecified()),
             noRestore,
             msbuildPath);
     }
 
+    public static int RunPackCommand(ParseResult parseResult)
+    {
+        var args = parseResult.GetValue(PackCommandParser.SlnOrProjectArgument)?.ToList() ?? new List<string>();
+
+        if (args.Count != 1)
+        {
+            Console.Error.WriteLine("Error: Only one .nuspec file can be packed at at time");
+            return 1;
+        }
+
+        var nuspecPath = args[0];
+
+        var packArgs = new PackArgs()
+        {
+            Path = nuspecPath,
+            BasePath = Path.GetDirectoryName(nuspecPath),
+            Logger = new NuGetConsoleLogger(),
+            Exclude = new List<string>()
+        };
+
+        var properties = parseResult.GetValue(PackCommandParser.PropertiesOption);
+        if (properties != null)
+        {
+            foreach (var prop in properties)
+            {
+                var split = prop.Split('=', 2);
+                if (split.Length == 2)
+                    packArgs.Properties[split[0]] = split[1];
+            }
+        }
```

Comment:
If you use the `CommonOptions.PropertiesOption` I mentioned above then you get this splitting/formatting for free already 👍